benchexec/test_core_assignment.py

```diff
@@ -100,601 +186,319 @@ def test_singleThread(self):
+    num_of_L3_regions = 8
+    num_of_hyperthreading_siblings = 2
+    use_hyperthreading = False
 
-    oneCore_assignment = [[x] for x in range(8)]
-    twoCore_assignment = [[0, 1], [2, 3], [4, 5], [6, 7]]
-    threeCore_assignment = [[0, 1, 2], [3, 4, 5]]
-    fourCore_assignment = [[0, 1, 2, 3], [4, 5, 6, 7]]
-    eightCore_assignment = [list(range(8))]
+    """                 x
 
-    def test_singleCPU_invalid(self):
-        self.assertInvalid(2, 5)
-        self.assertInvalid(5, 2)
-        self.assertInvalid(3, 3)
+            x                       x
 
+    x   x       x   x       x   x      x    x
 
-class TestCpuCoresPerRun_singleCPU_HT(TestCpuCoresPerRun_singleCPU):
-    ht = True
+    x-  x-      x-  x-      x-  x-     x-  x-
+    """
+    # expected results for different coreLimits
+    oneCore_assignment = [[x] for x in [0, 8, 2, 10, 4, 12, 6, 14]]
+    twoCore_assignment = [[0, 2], [8, 10], [4, 6], [12, 14]]
+    threeCore_assignment = [[0, 2, 4], [8, 10, 12]]
+    fourCore_assignment = [[0, 2, 4, 6], [8, 10, 12, 14]]
+    fiveCore_assignment = [[0, 2, 4, 6, 8]]
+    eightCore_assignment = [[0, 2, 4, 6, 8, 10, 12, 14]]
 
-    twoCore_assignment = [[0, 4], [1, 5], [2, 6], [3, 7]]
-    threeCore_assignment = [[0, 1, 4], [2, 3, 6]]
-    fourCore_assignment = [[0, 1, 4, 5], [2, 3, 6, 7]]
+    def test_fiveCoresPerRun(self):
+        self.t_unit_assertValid(5, self.fiveCore_assignment)
 
-    def test_halfPhysicalCore(self):
-        # Cannot run if we have only half of one physical core
-        self.assertRaises(
-            SystemExit,
-            _get_cpu_cores_per_run0,
-            1,
-            1,
-            True,
-            [0],
-            {0: [0, 1]},
-            {0: [0, 1]},
-        )
+    def test_invalid(self):
+        # coreLimit, num_of_threads
+        self.assertInvalid(2, 5)
+        self.assertInvalid(5, 2)
+        self.assertInvalid(3, 3)
 
 
-class TestCpuCoresPerRun_dualCPU_HT(TestCpuCoresPerRun):
-    cpus = 2
-    cores = 16
-    ht = True
-
-    oneCore_assignment = [
-        [x]
-        for x in [
-            0,
-            8,
-            1,
-            9,
-            2,
-            10,
-            3,
-            11,
-            4,
-            12,
-            5,
-            13,
-            6,
-            14,
-            7,
-            15,
-            16,
-            24,
-            17,
-            25,
-            18,
-            26,
-            19,
-            27,
-            20,
-            28,
-            21,
-            29,
-            22,
-            30,
-            23,
-            31,
-        ]
-    ]
+class Test_Topology_P1_NUMA2_L8_C16_T(TestCpuCoresPerRun):
+    num_of_cores = 16
+    num_of_packages = 1
+    num_of_NUMAs = 2
+    num_of_L3_regions = 8
+    num_of_hyperthreading_siblings = 2
+    use_hyperthreading = True
 
+    # expected results for different coreLimits
+    oneCore_assignment = [[x] for x in [0, 8, 2, 10, 4, 12, 6, 14]]
```

Comment:
I realized that some missunderstanding concerning the unit test arise from different core numbering an the way, the machine is built from the variables.  While there are 16 cores, halb of them are hyperthreading siblings. Since hyperthreading is off, our maximal number of threads is 8 for the one core assigment und 4 for the 2 core assigment.
I put the core id into the commets for clarification